app/src/features/surveys/components/CreateSurveyBlockDialog.tsx

```diff
@@ -0,0 +1,74 @@
+import CloseIcon from '@mui/icons-material/Close';
+import { Typography } from '@mui/material';
+import IconButton from '@mui/material/IconButton';
+import Snackbar from '@mui/material/Snackbar';
+import EditDialog from 'components/dialog/EditDialog';
+import { useState } from 'react';
+import BlockForm from './BlockForm';
+import { BlockYupSchema } from './SurveyBlockSection';
+interface ICreateBlockProps {
+  open: boolean;
+  onSave: (data: any) => void;
+  onClose: () => void;
+}
+
+const CreateSurveyBlockDialog: React.FC<ICreateBlockProps> = (props) => {
+  const { open, onSave, onClose } = props;
+  const [isSnackBarOpen, setIsSnackBarOpen] = useState(false);
+  const [blockName, setBlockName] = useState('');
+  return (
+    <>
+      <EditDialog
+        dialogTitle={'Add Block'}
+        dialogText={
+          'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam at porttitor sem. Aliquam erat volutpat. Donec placerat nisl magna, et faucibus arcu condimentum sed.'
+        }
+        open={open}
+        dialogLoading={false}
+        component={{
+          element: <BlockForm />,
+          initialValues: {
+            survey_block_id: null,
+            name: '',
+            description: ''
+          },
+          validationSchema: BlockYupSchema
+        }}
+        dialogSaveButtonLabel="Add Block"
+        onCancel={() => onClose()}
+        onSave={(formValues) => {
+          setBlockName(formValues.name);
+          setIsSnackBarOpen(true);
+          onSave(formValues);
+        }}
+      />
+      {/* This is done instead of the dialogContext because that causes the form to rerender, removing any changes the user makes */}
```

Comment:
Not entirely sure what this comment is suggesting. Is the Snackbar only included because we need to force a rerender of the form?

Reply:
Unless it's a requirement in the ticket, I think adding a snackbar is a bit out of place, since it's not used for any other controls on the Survey form (unless I'm mistaken)

Reply:
It is a requirement in the ticket. I might have to write a bit more but I was trying to explain why I didn't use the dialog context to render the snack bar.